build-ci/jenkins-stack-os-matrix.rst

```diff
@@ -0,0 +1,60 @@
+
+`stack-os-matrix`_ is the continuous integration (CI) job that DM developers must run to ensure tests pass before merging changes to any LSST EUPS Stack package.
+This page explains how to run the `stack-os-matrix`_ job and find build and test logs.
+
+For more information about the DM's Jenkins continuous integration service, see :doc:`jenkins`.
+
+Running a stack-os-matrix job
+=============================
+
+1. Open the `stack-os-matrix dashboard <stack-os-matrix>`_ in a browser directly or by clicking on the **stack-os-matrix** job from the main dashboard at https://ci.lsst.codes.
+
+2. Click on the **Run** button near the top of the page.
+
+3. Configure the run in the pop-up dialog:
+
+   - **Set the list of Git refs** (branch or tag names) to check out from Stack Git repositories.
+
+     For example, if you enter ``tickets/DM-2 tickets/DM-1``, the build system will attempt to check out the ``tickets/DM-2`` branch in Stack packages.
+     If a package does not have a ``tickets/DM-2`` branch, it will attempt to check out the ``tickets/DM-1`` branch.
+     If a package has neither branch, it falls back to checking out the ``master`` branch.
+
+   - **Set the list of EUPS packages** to build.
+     Use the default (``lsst_distrib``) to build and test your changes with a full Stack.
+     To improve build times you can instead specify the name of the package you are actively developing (see also **Skip Demo**, below).
+
+   - **Check the Skip Demo** option only if you are testing your package alone (in conjunction with specifying your package in the prior text box).
+
+     Before you merge a ticket branch, **you must** run a stack-os-matrix Job without **Skip Demo** enabled so that the full Stack is built and tested with your changes.
+
+     When the ``demo`` is run, the `lsst_ci`_ top-level package is automatically added to the build and the `lsst_dm_stack_demo`_ is run to test a simple :command:`processCcd.py`\ -based pipeline.
+
+4. Click the **Run** button in the dialog to start the build.
+
+Monitoring the run status
+=========================
+
+Your new job is added to the top of the table on the `stack-os-matrix dashboard <stack-os-matrix>`_.
+Click on the row to see a detail page for the run.
+
+The pipeline diagram at the top of the detail page shows the status of the build on each supported platform.
+Click on the icon for the platform you're interested in.
+
+The progress for each build step is listed on the run detail page.
+The most important build step is ``./buildbot-scripts/jenkins_wrapper.sh`` where the lsstsw rebuild is run.
+Click on that step to see build and test progress for each package.
+
+Viewing and downloading build logs
+==================================
+
+If a build failed, the most effective way to diagnose the problem is to look at the :file:`*.failed` files for individual packages.
+
+You can find these :file:`*.failed` files, along with build logs, under the **Artifacts** tab from the run detail pages.
```

Comment:
We could link to the Artifacts tab of an example run.  Is that just confusing things?  Or you could add a screen shot, but then I guess we would then need to keep it up to date.

I always have to go hunting for the Artifacts tab.

Reply:
If a test failed they should be looking at the test report in Jenkins before they go looking for artifacts for the test `.failed` file.